### ISSUE TITLE
TypeLibrary: Method to get members of certain accessibility

### ISSUE THREAD

**Issue by peter-r-g** [Facepunch/sbox-issues#2371]
### What it is?

Currently `TypeDescription.Members` is a collection of all members on a type regardless of their accessibility and instance/static definition. With `MemberDescription` having no way to distinguish what their accessibility is you cannot check and select members that are only public or only private/protected.

### What should it be?

Add `TypeDescription.GetMembers( bool public = true )`, it avoids exposing the `System.Reflection.BindingFlags` enum and gives enough functionality for people to distinguish between what they are looking for. The reason for the optional bool is that the default behavior of [Type.GetMembers](https://learn.microsoft.com/en-us/dotnet/api/system.type.getmembers?view=net-6.0) is to get all public members of a type.

Example implementation:
```cs
public MemberDescription[] GetMembers( bool public = true )
{
    var flags = BindingFlags.Instance | BindingFlags.Static;
    flags |= public ? BindingFlags.Public : BindingFlags.NonPublic;
    return TargetType.GetMembers( flags ).Select( member => MemberDescription.Create( this, member ) ).ToArray();
}
```
